src/node.zig

```diff
@@ -331,55 +465,147 @@ fn request(env: c.napi_env, info: c.napi_callback_info) callconv(.C) c.napi_valu
+        "Failed to get callback reference.",
+    ) catch return;
+    const scope = translate.scope(
+        env,
+        "Failed to get \"this\" for results callback.",
+    ) catch return;
     const globals_raw = translate.globals(env) catch return;
     const globals = globalsCast(globals_raw.?);
     const argc: usize = 2;
     var argv: [argc]c.napi_value = undefined;
 
-    // if (client_error != null) {
-    //     // TODO: determine message according to error
-    //     const napi_error = create_napi_error(env, "Client could not send batch.") catch return;
-    //     argv[0] = napi_error;
-    //     argv[1] = napi_undefined;
-    // } else {
-    //     const napi_results = encode_napi_results_array(env, @bitCast([]Result, results)) catch return;
-    //     argv[0] = napi_undefined;
-    //     argv[1] = napi_results;
-    // }
-
-    const napi_results = switch (operation) {
-        .reserved, .init => {
-            translate.throw(env, "Reserved operation.") catch return;
-        },
-        .create_accounts => encode_napi_results_array(CreateAccountsResult, env, results) catch return,
-        .create_transfers => encode_napi_results_array(CreateTransfersResult, env, results) catch return,
-        .commit_transfers => encode_napi_results_array(CommitTransfersResult, env, results) catch return,
-        .lookup_accounts => encode_napi_results_array(Account, env, results) catch return,
-    };
-    argv[0] = globals.napi_undefined;
-    argv[1] = napi_results;
+    if (results) |value| {
+        const napi_results = switch (operation) {
+            .reserved, .init => {
+                translate.throw(env, "Reserved operation.") catch return;
+            },
+            .create_accounts => encode_napi_results_array(
+                CreateAccountsResult,
+                env,
+                value,
+            ) catch return,
+            .create_transfers => encode_napi_results_array(
+                CreateTransfersResult,
+                env,
+                value,
+            ) catch return,
+            .commit_transfers => encode_napi_results_array(
+                CommitTransfersResult,
+                env,
+                value,
+            ) catch return,
+            .lookup_accounts => encode_napi_results_array(Account, env, value) catch return,
+        };
 
-    // TODO: add error handling (but allow JS function to throw an exception)
-    translate.call_function(env, scope, napi_callback, argc, argv[0..]) catch return;
+        argv[0] = globals.napi_undefined;
+        argv[1] = napi_results;
+    } else |err| {
+        argv[0] = create_client_error(env, err) catch {
+            translate.throw(env, "Failed to create Node Error.") catch return;
+        };
+        argv[1] = globals.napi_undefined;
+    }
+
+    translate.call_function(env, scope, napi_callback, argc, argv[0..]) catch {
+        translate.throw(env, "Failed to call JS results callback.") catch return;
+    };
```

Comment:
It's possible for the user's own JS callback function to throw an exception. If that happens, we'll land up here since the result checked by https://github.com/coilhq/tigerbeetle-node/blob/6b990b66f8696a72acb6ab96d5f7016611863fe7/src/translate.zig#L443 will then not be `.napi_ok`.

We should rather tweak L443 in `translate.zig` to expect and handle this as ok, so that we don't throw an exception here. If it's the user throwing an exception, then that's fine from our point of view, it's all within their application. For example, we may have called their callback, and that called something synchronously and so on until eventually that ended up with a user exception firing.